Plutarch/Reducible.hs

```diff
@@ -15,3 +15,8 @@ instance Reducible a => Reducible (Const a b) where
 
 instance Reducible a => Reducible (Identity a) where
   type Reduce (Identity a) = Reduce a
+
+newtype Flip f a b = Flip (f b a)
```

Comment:
I didn't do this on purpose. `Plutarch.Reducible` shouldn't export any helper types, since they're very often ad-hoc, and aren't relevant to the end-user anyway. They're completely internal and are not part of the interface of Plutarch. The `Reducible` instance is generically derived on my local machine, I'll push that later.

Reply:
It seems weird to have to copy/paste the same block of boilerplate code in 3 separate files in the same project. Though derivation helpers would indeed make it a bit better.

Reply:
it would only be one line then